crates/matrix-sdk-ui/tests/integration/timeline/mod.rs

```diff
@@ -831,8 +832,71 @@ async fn test_timeline_without_encryption_can_update() {
     assert_pending!(stream);
 }
 
+#[async_test]
+async fn test_timeline_receives_a_limited_number_of_events_when_subscribing() {
+    let room_id = room_id!("!foo:bar.baz");
+    let event_factory = EventFactory::new().room(room_id).sender(&ALICE);
+
+    let mock_server = MatrixMockServer::new().await;
+    let client = mock_server.client_builder().build().await;
+
+    mock_server.sync_joined_room(&client, room_id).await;
+
+    let event_cache = client.event_cache();
+    event_cache.subscribe().unwrap();
+
+    // The event cache contains 30 events.
+    event_cache
+        .add_initial_events(
+            room_id,
+            (0..30)
+                .map(|nth| {
+                    event_factory
+                        .text_msg("foo")
+                        .event_id(&EventId::parse(format!("$ev{nth}")).unwrap())
+                        .into_event()
+                })
+                .collect::<Vec<_>>(),
+            None,
+        )
+        .await
+        .unwrap();
+
+    let room = client.get_room(room_id).unwrap();
+
+    // The timeline is created.
+    let timeline = room.timeline().await.unwrap();
+    let (timeline_initial_items, mut timeline_stream) = timeline.subscribe().await;
+
+    // The timeline receives 20 initial values, not 30!
+    assert_eq!(timeline_initial_items.len(), 20);
+    assert_pending!(timeline_stream);
+
+    // To get the other, the timeline needs to paginate.
+
+    let _no_network_pagination =
+        mock_server.mock_room_messages().error500().never().mount_as_scoped().await;
```

Comment:
Again, you can remove it, because if the SDK attempts to do pagination, it'll hit a 404 and fail.